File: FRONT_END_APP/src/actions/TeamLeadService.js

import axios from 'axios'

const TEAM_LEAD_URL="http://localhost:8080/api/kanban_board/"
class TeamLeadService{
        teamLeadLogin(id, pass){
                return axios.get(TEAM_LEAD_URL+'teamLeadLogin/'+id+'/'+pass); 
        }
        getProjectByTeamLeadId(id){
                return axios.get(TEAM_LEAD_URL+'teamlead/'+id);
        }
        getAllTeamMembers(id){
            return axios.get(TEAM_LEAD_URL+'getteammembers/?projectId='+id);
        }
        updateStatus(id,status)
        {
                return axios.get(TEAM_LEAD_URL+'updatestatus?projectStatus='+status+'&projectId='+id);
        }
}

export default new TeamLeadService();

File: BACK_END_APP/kanban_board_management/src/main/java/com/cg/fsd4/kanban_board/services/TeamLeadService.java
package com.cg.fsd4.kanban_board.services;

import java.util.List;


import com.cg.fsd4.kanban_board.entity.ProjectEntity;
import com.cg.fsd4.kanban_board.entity.TeamLeadEntity;
import com.cg.fsd4.kanban_board.entity.TeamMemberEntity;
import com.cg.fsd4.kanban_board.exception.TeamLeadNotFoundException;


public interface TeamLeadService {
	
	
	public List<TeamLeadEntity> viewTeamLead()  throws TeamLeadNotFoundException;
	
	public TeamLeadEntity viewTeamLeadById(int teamLeadId)  throws TeamLeadNotFoundException;
	
	public TeamLeadEntity Login( int teamLeadId, String projectPassword)  throws TeamLeadNotFoundException;
	
	public ProjectEntity UpdateStatus(String status,int projectId)  throws TeamLeadNotFoundException;
	
	public List<TeamMemberEntity> getTeamMembers()  throws TeamLeadNotFoundException;
	
	public List<TeamLeadEntity> getAllTeamLead();
}
